AngularApp/src/app/modules/dashboard/tabs/tab-data/tab-data.component.html

```diff
@@ -1,4 +1,12 @@
-<div style="padding:20px">
+<div style="padding:20px" *ngIf="!dataSourceMode">
   <detector-view [detectorResponse]="detectorResponse" [error]="error" [startTime]="queryParamsService.startTime" [endTime]="queryParamsService.endTime">
   </detector-view>
 </div>
+
+<div style="padding:20px" *ngIf="dataSourceMode">    
```

Comment:
Lets not do this if else logic in one component.
Create a seprate tab component..   it will be much more flexible to add more stuff in future..

Reply:
On that note, you already have the data-sources component. Why not route to that directly from tab instead of coming here in Data Component

Reply:
@ShekharGupta1988  - My real idea for implementing it this way was to avoid making a call to get detector response *just* to get the data source info. I earlier thought of making it a standalone component but then we were making a getDetectorresponse() call just to see the Data Sources information so by doing it this way, I was trying to reduce the backend calls to just get detector data source information by re-using the Detector response. If you think that is not a big deal, I can just route it to a new component and code it that way. Let me know

Reply:
@puneetg1983 - are we making duplicate calls to backend if you separate it out.. 
If yes, Then thats a bug in itself..
We have a cache service that should avoid making duplicate calls .. just like when you click around detectors, we dont make duplicate calls for every click to same detector...

Reply:
I see, I wasnt aware of the caching service...I will validate and separate this out then